WordPress/src/main/java/org/wordpress/android/viewmodel/themes/ThemesViewModel.kt

```diff
@@ -0,0 +1,107 @@
+package org.wordpress.android.viewmodel.themes
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.ThemeStore
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T.THEMES
+import org.wordpress.android.viewmodel.ResourceProvider
+import org.wordpress.android.viewmodel.ScopedViewModel
+import org.wordpress.android.viewmodel.themes.ThemesViewModel.BottomSheetAction.Hide
+import org.wordpress.android.viewmodel.themes.ThemesViewModel.BottomSheetAction.Show
+import org.wordpress.android.viewmodel.themes.ThemesViewModel.BottomSheetUIState.Selection
+import org.wordpress.android.viewmodel.themes.ThemesViewModel.BottomSheetUIState.Selection.KeepCurrentHomepage
+import org.wordpress.android.viewmodel.themes.ThemesViewModel.BottomSheetUIState.Selection.UseThemeHomepage
+import javax.inject.Inject
+import javax.inject.Named
+
+class ThemesViewModel @Inject constructor(
+    private val themeStore: ThemeStore,
+    private val resourceProvider: ResourceProvider,
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+) : ScopedViewModel(mainDispatcher) {
+    private lateinit var site: SiteModel
+    private var isStarted = false
+
+    private val _bottomSheetUiState = MutableLiveData<BottomSheetUIState>()
+    val bottomSheetUiState: LiveData<BottomSheetUIState> = _bottomSheetUiState
+
+    private val _bottomSheetAction = MutableLiveData<BottomSheetAction>()
+    val bottomSheetAction: MutableLiveData<BottomSheetAction> = _bottomSheetAction
+
+    fun start(site: SiteModel) {
+        if (isStarted) return
+
+        this.site = site
+        _bottomSheetUiState.value = BottomSheetUIState()
+        isStarted = true
+    }
+
+    private fun currentViewState(): BottomSheetUIState = bottomSheetUiState.value!!
```

Comment:
We should avoid the the [not-null assertion operator](https://kotlinlang.org/docs/null-safety.html#the-operator) if it is not completely necessary. This might be safer if we use null checks or [safe calls](https://kotlinlang.org/docs/null-safety.html#safe-calls) instead.

Reply:
Done